tests/client/core/i18n/test_utils.js

```diff
@@ -387,4 +389,54 @@ describe('i18n utils', () => {
       assert.equal(result, undefined);
     });
   });
+
+  describe('makeI18n', () => {
+    let FakeJed;
+
+    before(() => {
+      FakeJed = class {
+        constructor(i18nData) {
+          return i18nData;
+        }
+      };
+    });
+
+    beforeEach(() => {
+      // FIXME: Our moment is not immutable so we reset it before each test.
+      // This is annoying to work around because of the locale `require()`s
+      // and it only affects tests so it'd be nice to fix but doesn't break
+      // anything.
+      moment.locale('en');
```

Comment:
If we were localising the moment instance you could check that `moment.locale()` is still `en` when setting the locale data to `fr` and remove this.